planner/core/logical_plan_builder.go

```diff
@@ -2216,7 +2245,16 @@ func getStatsTable(ctx sessionctx.Context, tblInfo *model.TableInfo, pid int64)
 	return statsTbl
 }
 
-func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName) (LogicalPlan, error) {
+func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, asName model.CIStr) (LogicalPlan, error) {
+	// Add comments style index hint like /*+ INDEX(t, idx1, idx2) */ into tn.IndexHints.
+	if hints := b.TableHints(); hints != nil {
+		for _, hint := range hints.indexHintList {
+			// Hints match both original table name and table alias.
+			if hint.tblName == tn.Name || hint.tblName == asName {
+				tn.IndexHints = append(tn.IndexHints, hint.indexHint)
```

Comment:
If the index does not exist, we should not raise errors as `use index` do.

Reply:
wow...

Reply:
PTAL again. orz